src/expression.rs

```diff
@@ -219,6 +219,517 @@ fn is_small(x: f64) -> bool {
     x.abs() < 1e-16
 }
 
+mod simplification {
+    use super::{
+        format_complex, is_small, Expression, ExpressionFunction, InfixOperator, MemoryReference,
+        PrefixOperator,
+    };
+    use egg::{define_language, rewrite as rw, Id, Language};
+    use ordered_float::OrderedFloat;
+    use std::{
+        ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Sub, SubAssign},
+        str::FromStr,
+    };
+    use symbolic_expressions::{parser, IntoSexp, Sexp};
+
+    pub(crate) fn run(expression: &Expression) -> Expression {
+        let sexp = expression.into_sexp();
+        let recexpr = sexp
+            .to_string()
+            .parse()
+            .expect("Expression->Sexp->RecExpr should work by design");
+        let runner = egg::Runner::default()
+            .with_expr(&recexpr)
+            .run(&make_rules());
+        let root = runner.roots[0];
+        let (_, best) = egg::Extractor::new(&runner.egraph, egg::AstSize).find_best(root);
+        let simpler_sexp = parser::parse_str(&best.to_string())
+            .expect("RecExpr implements to_sexp, so this parse should succeed");
+        let mut simpler_exp = Expression::from(&simpler_sexp);
+        if let Ok(number) = simpler_exp.evaluate(&Default::default(), &Default::default()) {
```

Comment:
For some reason, the rewrite rules don't cover turning, e.g. `(neg 1)` into `-1`. Trying to take care of this without the special case.